### PR TITLE
Add spaces after flow control statements

### PR DESCRIPTION
**Pull request by tgucio**
This PR adds missing spaces after flow control statements (if, switch, for etc.) found in some *.dart and *.java files.

A check is also added in the analyzer.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by bernaferrari** [comment]
I'm surprised there is not a lint that auto-catches this.

**Comment by pq** [comment]
yeah. This is a sticky one. As a general rule we don't have lints that report things that would be corrected by the formatter. In general this works fine but bites Flutter since you aren't using the formatter.

**Comment by tgucio** [comment]
Could be something for the analyzer (dev/bots/analyze.dart) I guess? @Hixie

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

But if you wanted to update `dev/bots/analyze.dart` to catch this that would be great! The trick is making sure there's no false positives.

**Review by Hixie — APPROVED** [review]
LGTM

**Comment by tgucio** [comment]
@Hixie I'll take a stab at adding a check in the analyzer then - I guess it's OK to include as part of this PR.

**Comment by tgucio** [comment]
@Hixie Analyzer has been updated in case you'd like to TAL. The failing Windows and Android tests are unrelated (they appear to be flaky these days). There's also a failure in Google testing though that I guess we'd need a bit of help to look into.

File: dev/bots/analyze.dart [hunk 1]
```diff
@@ -1040,6 +1043,38 @@ Future<void> verifyNoTrailingSpaces(String workingDirectory, { int minimumMatche
   }
 }
 
+final RegExp _flowControlStatementWithoutSpace = RegExp(r'(^|[ \t])(if|switch|for|do|while|catch)\(');
```
Comment (Hixie) [inline dev/bots/analyze.dart:1046]: You can probably use `\b` instead of `(^|[ \t])` ?

Comment (Hixie) [reply dev/bots/analyze.dart:1046]: if you do want to use `(^|[ \t])` then you'll need to set `multiLine: true` to the RegExp constructor.

**Comment by Hixie** [comment]
This is great!

(If you want to get really inception-like, there's a test for the analyze script in `dev/bots/test/analyze_test.dart`, but I don't think it's critical to add a test for this test.)

(The "Google testing" error is almost certainly a flake, this script is not used in Google. If you rebase and repush it should rerun the tests.)

File: dev/bots/analyze.dart (hunk 1 above)
Comment (tgucio) [reply dev/bots/analyze.dart:1046]: Ah, yes, \b works here - thanks.

Comment (tgucio) [reply dev/bots/analyze.dart]: Actually I'll revert to `(^|[ \t])` as with `\b` the following async `catch()` method invocation in Java now matches the RegExp:

```
await serviceWorkerLoader.loadServiceWorker(serviceWorker).catch(e => {
```

I'll add `multiLine: true` though I guess it shouldn't be strictly needed as the input is split into lines already via `readAsLinesSync()`;

**Pull request merged**